datamap/mv/core/src/main/scala/org/apache/carbondata/mv/datamap/MVAnalyzerRule.scala

```diff
@@ -82,9 +89,19 @@ class MVAnalyzerRule(sparkSession: SparkSession) extends Rule[LogicalPlan] {
         }
         Aggregate(grp, aExp, child)
     }
-    val catalog = DataMapStoreManager.getInstance().getDataMapCatalog(dataMapProvider,
-      DataMapClassProvider.MV.getShortName).asInstanceOf[SummaryDatasetCatalog]
-    if (needAnalysis && catalog != null && isValidPlan(plan, catalog)) {
+    val summaryDataSetCatalog: java.util.List[SummaryDatasetCatalog] = new util
+    .ArrayList[SummaryDatasetCatalog]()
+    dataMapProvider.asScala.foreach(dataMapProvider =>
+      summaryDataSetCatalog.add(
+        DataMapStoreManager.getInstance().getDataMapCatalog(dataMapProvider,
+          dataMapProvider.getDataMapSchema.getProviderName).asInstanceOf[SummaryDatasetCatalog]))
+    var catalog: SummaryDatasetCatalog = null
+    summaryDataSetCatalog.asScala.foreach { summaryDataSetCatalog =>
+      if (summaryDataSetCatalog != null && isValidPlan(plan, summaryDataSetCatalog)) {
+        catalog = summaryDataSetCatalog
```

Comment:
if more than one SummaryDatasetCatalog is valid plan from catalogList (like mv and mv timeseries), here we are updating with the last one. please check